nifi-api/src/main/java/org/apache/nifi/parameter/SensitiveParameterProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.parameter;
+
+/**
+ * A base interface for all sensitive <code>ParameterProvider</code>s
+ */
+public interface SensitiveParameterProvider extends ParameterProvider {
```

Comment:
I believe I was initially going for strong type enforcement of parameter provider sensitivity here, but on reconsideration I agree.  Further, I'd like to go in the direction of making `ParameterProviderNode` responsible for enforcing/setting sensitivity based on how a ParameterProvider is referenced rather than exposing a method like this to the `ParameterProvider` interface.